src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,9 +32,26 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+
         this.value = trimmedAddress;
+        createAddress(trimmedAddress);
     }
 
+public void createAddress (String trimmedAddress) throws IllegalValueException
+{
+
+         String[] splitAddress = trimmedAddress.split(", ");
+
+         if (splitAddress.length  != 4) {
+             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+         }
+             Block block = new Block(splitAddress[0]);
+         Street street = new Street(splitAddress[1]);
+         Unit unit = new Unit(splitAddress[2]);
+         PostalCode postalCode = new PostalCode(splitAddress[3]);
```

Comment:
Don't use magic numbers. Also, I don't see how the user will get all these new information.